build-tools-internal/src/main/java/org/elasticsearch/gradle/internal/dra/DraResolvePlugin.java

```diff
@@ -0,0 +1,117 @@
+        this.providerFactory = providerFactory;
+        this.buildLayout = buildLayout;
+        this.snapshotRepositoryPrefix = providerFactory.systemProperty("dra.artifacts.url.repo.snapshot.prefix");
+        this.releaseRepositoryPrefix = providerFactory.systemProperty("dra.artifacts.url.repo.release.prefix");
+    }
+
+    @Override
+    public void apply(Project project) {
+        if (providerFactory.systemProperty(USE_DRA_ARTIFACTS_FLAG).isPresent()) {
+            resolveBuildIdProperties().get().forEach((key, buildId) -> {
+                configureDraRepository(
+                    project,
+                    "dra-snapshot-artifacts-" + key,
+                    key,
+                    buildId,
+                    snapshotRepositoryPrefix.orElse("https://artifacts-snapshot.elastic.co/"),
+                    ".*SNAPSHOT"
+                );
+                configureDraRepository(
+                    project,
+                    "dra-release-artifacts-" + key,
+                    key,
+                    buildId,
+                    releaseRepositoryPrefix.orElse("https://artifacts.elastic.co/"),
+                    "^(.(?!SNAPSHOT))*$"
+                );
+            });
+        }
+    }
+
+    private void configureDraRepository(
+        Project project,
+        String repositoryName,
+        String draKey,
+        String buildId,
+        Provider<String> repoPrefix,
+        String includeVersionRegex
+    ) {
+        project.getRepositories().ivy(repo -> {
+            repo.setName(repositoryName);
+            repo.setUrl(repoPrefix.get());
+            repo.patternLayout(patternLayout -> {
+                patternLayout.artifact(
+                    String.format("/%s/%s/downloads/%s/[module]-[revision]-[classifier].[ext]", draKey, buildId, draKey)
+                );
+                patternLayout.artifact(
+                    String.format("/%s/%s/downloads/%s/[module]/[module]-[revision]-[classifier].[ext]", draKey, buildId, draKey)
+                );
+            });
+            repo.metadataSources(metadataSources -> metadataSources.artifact());
+            repo.content(repositoryContentDescriptor -> repositoryContentDescriptor.includeVersionByRegex(".*", ".*", includeVersionRegex));
+        });
+    }
+
+    private Provider<Map<String, String>> resolveBuildIdProperties() {
+        return providerFactory.systemPropertiesPrefixedBy(DRA_ARTIFACTS_DEPENDENCY_PREFIX)
+            .map(
+                stringStringMap -> stringStringMap.entrySet()
+                    .stream()
+                    .collect(
+                        Collectors.toMap(entry -> entry.getKey().substring(DRA_ARTIFACTS_DEPENDENCY_PREFIX.length() + 1), Entry::getValue)
+                    )
+            )
+            .zip(providerFactory.provider(() -> {
+                Properties draProperties = new Properties();
+                File buildIdsFile = new File(buildLayout.getRootDirectory(), "buildIds.properties");
```

Comment:
In what scenario would we get these from a properties file vs system properties? Is it necessary to support this use case?

Reply:
I initially started with that and then moved to passing in via sys props. I'll delete that bit here again to keep it simple. initially thought it might be handy but not worth to keep it I guess

Reply:
removed